install.sh

```diff
@@ -94,133 +104,125 @@ esac
+    echo "Adding new information on ${1} to device.json..."
     sed -i device.json -e '/^  "installed_packages": \[$/s/$/\
     {\
-      "name": "'$1'",\
-      "version": "'$2'"\
+      "name": "'"${1}"'",\
+      "version": "'"${2}"'"\
     }/'
   fi
 }
 
 # create the device.json file if it doesn't exist
-cd $CREW_CONFIG_PATH
+cd "${CREW_CONFIG_PATH}"
 if [ ! -f device.json ]; then
   echo "Creating new device.json..."
   echo '{' > device.json
-  echo '  "architecture": "'$architecture'",' >> device.json
+  echo '  "architecture": "'"${ARCH}"'",' >> device.json
   echo '  "installed_packages": [' >> device.json
   echo '  ]' >> device.json
   echo '}' >> device.json
 fi
 
 # extract, install and register packages
-for i in `seq 0 $((${#urls[@]} - 1))`; do
-  url=${urls[$i]}
-  sha256=${sha256s[$i]}
-  tarfile=`basename $url`
-  name=${tarfile%%-*}   # extract string before first '-'
-  rest=${tarfile#*-}    # extract string after first '-'
-  version=`echo $rest | sed -e 's/-chromeos.*$//'`
+for i in $(seq 0 $((${#urls[@]} - 1))); do
+  url="${urls["${i}"]}"
+  sha256="${sha256s["${i}"]}"
+  tarfile="$(basename ${url})"
+  name="${tarfile%%-*}"   # extract string before first '-'
+  rest="${tarfile#*-}"    # extract string after first '-'
+  version="$(echo ${rest} | sed -e 's/-chromeos.*$//')"
                         # extract string between first '-' and "-chromeos"
 
-  download_check $name $url $tarfile $sha256
-  extract_install $name $tarfile
-  update_device_json $name $version
+  download_check "${name}" "${url}" "${tarfile}" "${sha256}"
+  extract_install "${name}" "${tarfile}"
+  update_device_json "${name}" "${version}"
 done
 
-# download, prepare and install chromebrew
-cd $CREW_LIB_PATH
-rm -rf crew lib packages
-curl -# -o crew $URL/crew
-chmod +x crew
-rm -f $CREW_PREFIX/bin/crew
-ln -s `pwd`/crew $CREW_PREFIX/bin
-#install crew library
-mkdir -p $CREW_LIB_PATH/lib
-cd $CREW_LIB_PATH/lib
-curl -# -o package.rb $URL/lib/package.rb
-curl -# -o package_helpers.rb $URL/lib/package_helpers.rb
-
-# Making GCC act like CC (For some npm packages out there, only required for gcc)
-#rm -f $CREW_PREFIX/bin/cc
-#ln -s $CREW_PREFIX/bin/gcc $CREW_PREFIX/bin/cc
-
-# package gcc7 has already made symbolic links for cc and gcc, no action required here
+# create symlink to 'crew' in ${CREW_PREFIX}/bin/
+rm -f "${CREW_PREFIX}/bin/crew"
+ln -s "../lib/crew/crew" "${CREW_PREFIX}/bin/"
 
 # prepare sparse checkout .rb packages directory and do it
-cd $CREW_LIB_PATH
+cd "${CREW_LIB_PATH}"
 rm -rf .git
 git init
-git remote add -f origin https://github.com/$OWNER/$REPO.git
+git remote add -f origin "https://github.com/${OWNER}/${REPO}.git"
 git config core.sparsecheckout true
 echo packages >> .git/info/sparse-checkout
 echo lib >> .git/info/sparse-checkout
 echo crew >> .git/info/sparse-checkout
-git fetch origin master
-git reset --hard origin/master
+git fetch origin "${BRANCH}"
+git reset --hard origin/"${BRANCH}"
```

Comment:
Where is `${BRANCH}` getting the value?

Reply:
Up at the top where most of the variables are set.

Reply:
I was having some troubles setting `${BRANCH}` when I was testing out `linux_sources`. This fixes that.